firebase/package.json

```diff
@@ -5,28 +5,36 @@
   },
   "private": true,
   "scripts": {
-    "lint": "eslint '**/*.ts'",
+    "postinstall": "run-p install:*",
+    "install:functions": "cd src/functions && npm i",
     "watch": "tsc -w",
-    "test": "npm run firestore-rules-test",
-    "firebase-deploy": "firebase deploy",
-    "firestore-deploy": "firebase deploy --only firestore",
-    "firestore-rules-test": "firebase emulators:exec --only firestore 'mocha -r ts-node/register --file \"./test/firestore-rules/setup.ts\" --recursive \"./test/firestore-rules/**/*.spec.ts\" --timeout 10000 --exit'",
-    "functions-fetch-env": "firebase functions:config:get > src/functions/.runtimeconfig.json",
-    "functions-build": "npm run lint && tsc -p tsconfig.functions.json",
-    "functions-serve": "npm run functions-build && npm run watch | firebase emulators:start --only auth,functions,firestore --inspect-functions",
-    "functions-deploy": "firebase deploy --only functions",
-    "dummy-script": "ts-node src/scripts/dummy-script.ts"
+    "lint": "run-p lint:*",
+    "lint:core": "eslint 'src/core/**/*.ts'",
+    "lint:functions": "eslint 'src/functions/**/*.ts'",
+    "lint:scripts": "eslint 'src/scripts/**/*.ts'",
+    "lint:test": "eslint 'test/**/*.ts'",
+    "deploy": "firebase deploy",
+    "deploy:firestore": "firebase deploy --only firestore",
+    "test": "run-p test:*",
+    "test:firestore-rules": "firebase emulators:exec --only firestore 'mocha -r ts-node/register --file \"./test/firestore-rules/setup.ts\" --recursive \"./test/firestore-rules/**/*.spec.ts\" --timeout 10000 --exit'",
+    "build": "run-p build:*",
+    "build:functions": "tsc -p tsconfig.functions.json",
+    "functions:serve": "run-s --print-label build:functions functions:start-emulators watch",
+    "functions:fetch-env": "firebase functions:config:get > src/functions/.runtimeconfig.json",
+    "functions:start-emulators": "firebase emulators:start --only auth,functions,firestore --inspect-functions",
+    "scripts:dummy-script": "ts-node src/scripts/dummy-script.ts"
   },
   "devDependencies": {
     "@firebase/rules-unit-testing": "^1.x",
-    "@types/mocha": "^9.0.0",
+    "@types/mocha": "^9.x",
     "@types/node": "^14.x",
     "@typescript-eslint/eslint-plugin": "^4.x",
     "@typescript-eslint/parser": "^4.x",
     "eslint": "^7.x",
     "eslint-config-prettier": "^8.x",
     "eslint-plugin-prettier": "^3.x",
-    "mocha": "^9.0.3",
+    "mocha": "^9.x",
+    "npm-run-all": "^4.x",
     "prettier": "^2.x",
     "ts-node": "^10.x"
```

Comment:
What about locking dependencies?

Reply:
I'll lock them for sure, but in a future PR pointing to `server` branch